92-AppendixC.Rmd

```diff
@@ -0,0 +1,312 @@
+response <- 
+  c("strongly agree", "agree", "agree", "disagree")
+
+response_levels <-
+  c("strongly agree", "agree", "disagree", "strongly disagree")
+
+factors <- factor(response, levels = response_levels)
+
+factors
+```
+
+Factors are integer vectors, though they may look like character strings. We can confirm by looking at the vector's structure:
+
+```{r}
+#| label: readr-factor-view
+glimpse(factors)
+```
+
+R's factors differ from Stata, SPSS, or SAS' labeled vectors. However, we can convert labeled variables into factors using the `as_factor()` function.
+
+```{r}
+#| label: readr-factor-create
+anes_dta %>% 
+  transmute(V200002 = as_factor(V200002))
+```
+
+The `as_factor()` function can be applied to all columns in a data frame or individual ones. Below, we convert all `<dbl+lbl>` columns into factors.
+
+```{r}
+#| label: readr-factor-glimpse
+anes_dta_factor <-
+  anes_dta %>% 
+  as_factor()
+
+anes_dta_factor %>% 
+  select(1:6) %>% 
+  glimpse()
+```
+
+#### Option 2: Strip the labels{-}
+
+The second option is to remove the labels altogether, converting the labeled data into a regular R data frame. To remove, or 'zap' the labels from our tibble, we can use the {haven} package's `zap_label()` and `zap_labels()` functions. This approach removes the labels but retains the data values in their original form.
+
+Use `zap_label()` to remove the variable labels but retain the value labels:
+
+```{r}
+#| label: readr-zaplabel
+zap_label(anes_dta) %>% 
+  select(1:6) %>% 
+  glimpse()
+```
+
+To remove the value labels, use `zap_labels()`. Notice the previous `<dbl+lbl>` columns are now `<dbl>`.
+
+```{r}
+#| label: readr-zaplabels
+zap_labels(anes_dta) %>% 
+  select(1:6) %>% 
+  glimpse()
+```
+
+While it is important to convert labeled datasets into regular R data frames for working in R, the labels themselves often contain valuable information that provide context and meaning to the survey variables. To aid with interpretability and documention, consider creating a data dictionary from the labeled dataset. A data dictionary is a reference document that provides detailed information about the variables and values of a survey.
+
+The {labelled} package offers a convenient function, `generate_dictionary()`, that creates data dictionaries directly from a labeled dataset. This function extracts variable labels, value labels, and other metadata and organizes them into a structured document that we can browse and reference throughout our analysis. 
+
+Let's create a data dictionary from the ANES Stata dataset as an example:
+
+```{r}
+#| label: readr-dictionary-create
+library(labelled)
+
+dictionary <- generate_dictionary(anes_dta)
+```
+
+Once we've generated the data dictionary, we can take a look at the `V200002` variable and see the label, column type, number of missing entries, and associated values.
+
+```{r}
+#| label: readr-dictionary-view
+dictionary %>% 
+  filter(variable == "V200002")
+```
+
+### Labeled missing data values
```

Comment:
Would love to see some examples.  Does this get read into R as `na_A_` for example?  Like we have `na_real_`?  Some examples of what these look like would be helpful.  If ANES data doesn't have this, can we find an example to show it?  Is there data from the naniar package that has some examples we can just show (without explaining what the data is)?